### PR TITLE
added PROVIDE_ACTION_TARGETS config, False by default

### PR DESCRIPTION
**Pull request by kywch**
* fixed scripted agents to resume sell, buy, use
* added valid mask for action.Price to ActionTargets in Observation (replacing None)
* added ActionTargets to Env.observation_space()

---
Here's more

* added test_monkey_actions to issue random (valid) actions within the action space
* rewrote env._process_actions() to process both neural and scripted agents actions by enabling arg deserialization for all actions --> and I could delete a lot of redundant code, yay!
* saw immortal agents lurking around the map edges and causing problems --> fixed it by not letting agents walk on lava. However, due to this our beloved LavaAgent (https://neuralmmo.github.io/build/html/rst/tutorial.html#icon-scripted-api) doesn't work anymore. How do we want to handle this?
* brought back two disabled performance tests

### CODE REVIEW

File: nmmo/core/env.py [hunk 1]
```diff
@@ -93,6 +97,17 @@ def action_space(self, agent):
 
     actions = {}
     for atn in sorted(nmmo.Action.edges(self.config)):
+
+      # check if each system is enabled in config
```
Comment (daveey) [inline nmmo/core/env.py]: how about we define action.enabled(config) and just check there?

File: tests/test_eventlog.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
+# pylint: disable=all
```
Comment (daveey) [inline tests/test_eventlog.py]: i think this snuck into the PR

**Review by daveey — APPROVED** [review]
Looks great, feel free to merge after fixing

File: tests/test_eventlog.py (hunk 2 above)
Comment (kywch) [reply tests/test_eventlog.py]: oh shoot, I took it out!

File: nmmo/core/env.py (hunk 1 above)
Comment (kywch) [reply nmmo/core/env.py]: Thanks, David! I added action.enabled(config) and got rid of the long if statement.

**Pull request merged**